Monika After Story/game/definitions.rpy

```diff
@@ -1453,6 +1478,55 @@ python early:
             if ev.action in Event.ACTION_MAP:
                 Event._performAction(ev, **kwargs)
 
+        @staticmethod
+        def checkUndoActionRules():
+            """
+            Checks the undo action rules of the events in the undo action rule dict
+            """
+
+            for ev_label in persistent._mas_undo_action_rules.iterkeys():
+                if MASUndoActionRule.evaluate_rule(ev_label):
+                    Event._undoEVAction(ev_label)
+
+        @staticmethod
+        def _undoEVAction(ev_label):
```

Comment:
that's still not an excuse to have ur code like this. This is about proper organization of where to put code, not about how we store data. You can  see that none of the other `Event` functions do what you do, because thats not the right way to organize this. I've made the changes myself cause its clear you do not understand how to organize this.